cfme/tests/infrastructure/test_kubevirt.py

```diff
@@ -3,23 +3,94 @@
 import pytest
 
 from cfme.infrastructure.provider.kubevirt import KubeVirtProvider
+from cfme.provisioning import do_vm_provisioning
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.generators import random_vm_name
+from cfme.utils.log import logger
 from cfme.utils.update import update
 
+
 pytestmark = [
-    pytest.mark.provider([KubeVirtProvider], scope="module")
+    pytest.mark.provider([KubeVirtProvider]),
+    pytest.mark.usefixtures('setup_provider')
 ]
 
 
-def test_k6t_provider_crud(provider):
-    """Tests provisioning via PXE
+@pytest.fixture
+def vm_name():
+    vm_name = random_vm_name('k6tvm')
+    return vm_name
+
+
+@pytest.fixture
+def prov_data(vm_name):
+    return {'catalog': {'vm_name': vm_name}}
```

Comment:
Again, I don't think this really needs to be a fixture. temp_vm is useful as a fixture so you can do cleanup, and because its providing a complex action.

Otherwise within the test, you can just call random_vm_name and assign it to a variable to pass it around, and define this dictionary to update with the custom parametrized data.

Reply:
changed